game/world/objects/npc.cpp

```diff
@@ -409,7 +409,7 @@ void Npc::setDirectionY(float rotation) {
   if(rotation<-90)
     rotation = -90;
   rotation = std::fmod(rotation,360.f);
-  if(!mvAlgo.isSwim())
+  if(!mvAlgo.isSwim() && !(interactive()!=nullptr && interactive()->isLadder()))
```

Comment:
I don't mind such code(why only ladder tho?), but is this case should be already handled in `PlayerControl`

Reply:
No if at least one of  `isSwim()` or `isLadder()` is true the statement returns false and `angleY` is set. I need it to have access to the tilt factor in `tickMobsi` routine in `movealgo.cpp`.

Reply:
Right, but what this code is for? `angleY` is set to be a swimming rotation, I don't see how this fit into laddering

Reply:
For tilted ladders. I need to know how much the character has to move forward and when to apply it.

Reply:
Oh, now I think I'm getting it. Yep `angleY` is correct tool in such case.
Curious: is there any titled ladder in the game? Gothic1?